nixos/modules/services/mail/public-inbox.nix

```diff
@@ -0,0 +1,569 @@
+          type = types.submodule {
+            freeformType = iniAttrs;
+            options.spamcheck = mkOption {
+              type = with types; enum [ "spamc" "none" ];
+              default = "none";
+              description = ''
+                If set to spamc, ${manref "public-inbox-watch" 1} will filter spam
+                using SpamAssassin.
+              '';
+            };
+            options.watchspam = mkOption {
+              type = with types; nullOr str;
+              default = null;
+              example = "maildir:/path/to/spam";
+              description = ''
+                If set, mail in this maildir will be trained as spam and
+                deleted from all watched inboxes
+              '';
+            };
+          };
+        };
+        options.coderepo = mkOption {
+          default = {};
+          description = "code repositories";
+          type = types.submodule {
+            freeformType = types.attrsOf (types.submodule {
+              freeformType = types.either (types.attrsOf iniAtom) iniAtom;
+              options.cgitUrl = mkOption {
+                type = types.str;
+                description = "URL of a cgit instance";
+              };
+              options.dir = mkOption {
+                type = types.str;
+                description = "Path to a git repository";
+              };
+            });
+          };
+        };
+      };
+    };
+    openFirewall = mkEnableOption "opening the firewall when using a port option";
+  };
+  config = mkIf cfg.enable {
+    assertions = [
+      { assertion = config.services.spamassassin.enable || !useSpamAssassin;
+        message = ''
+          public-inbox is configured to use SpamAssassin, but
+          services.spamassassin.enable is false.  If you don't need
+          spam checking, set `services.public-inbox.settings.publicinboxmda.spamcheck' and
+          `services.public-inbox.settings.publicinboxwatch.spamcheck' to null.
+        '';
+      }
+      { assertion = cfg.path != [] || !useSpamAssassin;
+        message = ''
+          public-inbox is configured to use SpamAssassin, but there is
+          no spamc executable in services.public-inbox.path.  If you
+          don't need spam checking, set
+          `services.public-inbox.settings.publicinboxmda.spamcheck' and
+          `services.public-inbox.settings.publicinboxwatch.spamcheck' to null.
+        '';
+      }
+    ];
+    services.public-inbox.settings =
+      filterAttrsRecursive (n: v: v != null) {
+      publicinbox = mapAttrs (n: filterAttrs (n: v: n != "description")) cfg.inboxes;
+    };
+    users = {
+      users.public-inbox = {
+        # Use runCommand instead of linkFarm,
+        # because Postfix rejects .forward if it's a symlink.
+        home = pkgs.runCommand "public-inbox-home" {} (''
+          install -D -p ${environment.PI_CONFIG} $out/.public-inbox/config
+          ln -s ${stateDir}/emergency $out/.public-inbox/emergency
```

Comment:
I don't think the `home` option is intended to be used for such ever-changing stateless values. Is it necessary to populate the home directory? Or is there perhaps a command line option where this directory could be passed instead?

Reply:
I've changed to use `/var/lib/public-inbox` as `home`.